dnf5-plugins/config-manager_plugin/addrepo.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef DNF5_COMMANDS_CONFIG_MANAGER_CONFIG_MANAGER_ADDREPO_HPP
+#define DNF5_COMMANDS_CONFIG_MANAGER_CONFIG_MANAGER_ADDREPO_HPP
+
+#include <dnf5/context.hpp>
+
+#include <filesystem>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace dnf5 {
+
+class ConfigManagerAddRepoCommand : public Command {
+public:
+    explicit ConfigManagerAddRepoCommand(Context & context) : Command(context, "addrepo") {}
+    void set_argument_parser() override;
+    void configure() override;
+
+private:
+    void add_repo_from_repofile(const std::string & url, const std::filesystem::path & repo_dir);
+    void create_repo(const std::string & url, const std::filesystem::path & repo_dir);
+    void test_filepath(const std::filesystem::path & dest_path) const;
+    void test_if_ids_not_already_exist(
```

Comment:
May I ask you for a description of the method. The name is quite self descriptive but additional information how arguments are used might be helpful.

Reply:
I added doc string